include/sound/sof/info.h

```diff
@@ -111,7 +111,7 @@ struct sof_ipc_cc_version {
 
 	char name[16]; /* null terminated compiler name */
 	char optim[4]; /* null terminated compiler -O flag value */
-	char desc[]; /* null terminated compiler description */
+	char desc[24]; /* null terminated compiler description */
```

Comment:
what exactly is the problem here? this was added by firmware folks so what issue is this about?
Also the commit message is far from clear and riddled with typos.

Reply:
@plbossart Ok, this is https://github.com/thesofproject/sof/issues/2507 
The new commit is better but the root issue still becomes clear.

Reply:
why is '24' a suitable value? where does this come from and why not round to 32?

Also the commit description remains cryptic, I barely understand it despite reading the SOF issue. Please put yourself in the shoes of a maintainer and explain why this is happens and why this the 'right' fix. My understanding is that this is just a work-around.

Reply:
Ack @plbossart . I asked this in the FW PR: https://github.com/thesofproject/sof/pull/2522#issuecomment-601596467 -- let's conclude and merge then.